src/app/projects/project-details/project-details.component.ts

```diff
@@ -68,4 +77,30 @@ export class ProjectDetailsComponent implements OnInit {
     }
     selection.removeAllRanges();
   }
+
+  private updateTitle() {
+    if (!this.project) {
+      switch (this.activeTabIndex) {
+        case 0:
+          return this.titleService.setTitle(`Builds - Wharf`);
+        case 1:
+          return this.titleService.setTitle(`Configuration - Wharf`);
+        case 2:
+          return this.titleService.setTitle(`Schedule - Wharf`);
+        default:
+          return this.titleService.setTitle(`Project - Wharf`);
+      }
+    }
+
+    switch (this.activeTabIndex) {
+      case 0:
+        return this.titleService.setTitle(`Builds - ${this.project.name} - Wharf`);
+      case 1:
+        return this.titleService.setTitle(`Configuration - ${this.project.name} - Wharf`);
+      case 2:
+        return this.titleService.setTitle(`Schedule - ${this.project.name} - Wharf`);
+      default:
+        return this.titleService.setTitle(`${this.project.name} - Wharf`);
+    }
+  }
```

Comment:
!nit: Suggest removing the return statements. I find them confusing, as the `setTitle` doesn't return anything. Placing them after the switch statements would be semantically equivalent.

nit: Would rather see the text "Loading" than no project name.

Such as

```ts
`Loading... - Wharf`

`Builds - ${this.project.name} - Wharf`
`Configuration - ${this.project.name} - Wharf`
`Schedule - ${this.project.name} - Wharf`
`${this.project.name} - Wharf`
```

Reply:
Changed to use `Loading...` for the case when the project data hasn't been fetched yet, as suggested.
Changed to use if/else instead of switch statement.